includes/class-wc-payments-order-service.php

```diff
@@ -138,15 +150,14 @@ public function mark_payment_completed( $order, $intent_id, $intent_status, $cha
 	 * @return void
 	 */
 	public function mark_payment_failed( $order, $intent_id, $intent_status, $charge_id, $message = '' ) {
-		if ( $order->has_status( [ Order_Status::FAILED ] )
-			|| 'failed' === $order->get_meta( '_intention_status' )
```

Comment:
I couldn't find anywhere that set the `_intention_status` to `failed`, so I removed this check.